tests/test_main.py

```diff
@@ -27,3 +33,62 @@ def test_expand_abbreviations():
 
     expanded_template = expand_abbreviations(template, {})
     assert is_repo_url(expanded_template) is True
+
+
+@pytest.fixture(scope='session')
+def user_dir(tmpdir_factory):
+    return tmpdir_factory.mktemp('user_dir')
+
+
+@pytest.fixture(scope='session')
+def user_config_data(user_dir):
+    cookiecutters_dir = user_dir.mkdir('cookiecutters')
+    replay_dir = user_dir.mkdir('cookiecutter_replay')
+
+    return {
+        'cookiecutters_dir': str(cookiecutters_dir),
+        'replay_dir': str(replay_dir),
+        'default_context': {}
+    }
+
+
+@pytest.fixture(scope='session')
+def user_config_file(user_dir, user_config_data):
+    config_file = user_dir.join('config')
+    config_file.write(json.dumps(user_config_data))
+    return str(config_file)
+
+
+@pytest.fixture
+def template_url():
+    return 'https://github.com/pytest-dev/cookiecutter-pytest-plugin.git'
+
+
+@pytest.fixture
+def output_dir(tmpdir):
+    return tmpdir.mkdir('output')
+
+
+def test_cookiecutter_repository_url_should_clone(
+        mocker, template_url, output_dir, user_config_file, user_config_data):
+    mock_clone = mocker.patch(
+        'cookiecutter.main.clone',
+        return_value='tests/fake-repo-tmpl',
+        autospec=True
+    )
+
+    project_dir = cookiecutter(
+        template_url,
+        no_input=True,
+        output_dir=str(output_dir),
+        config_file=user_config_file
+    )
+
+    mock_clone.assert_called_once_with(
+        repo_url=template_url,
+        checkout=None,
+        clone_to_dir=user_config_data['cookiecutters_dir'],
+        no_input=True
+    )
+
+    assert os.path.isdir(project_dir)
```

Comment:
This test is a great start. If we remove the integration tests, I suppose we ought to unit test the `cookiecutter()` function more thoroughly. 

To address this, I noticed that there are already other unit tests covering lines of `cookiecutter()` in cookiecutter/tests/replay/test_replay.py. Perhaps we can move them here to start, and then see what lines still need tests along the lines of the one you've written here.